pkg/sdkserver/sdkserver.go

```diff
@@ -267,12 +238,69 @@ func (s *SDKServer) Run(ctx context.Context) error {
 			}
 		}
 	}()
-	defer s.server.Close() // nolint: errcheck
 
 	s.workerqueue.Run(ctx, 1)
 	return nil
 }
 
+// newHealthzServer creates the /healthz and /gshealthz handlers. We pass the
+// initialDelay directly to avoid races retrieving it from s.health, especially
+// in unit tests.
+func (s *SDKServer) newHealthzServer(initialDelay time.Duration) *http.Server {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("ok"))
+		if err != nil {
+			s.logger.WithError(err).Error("could not send ok response on healthz")
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	})
+	mux.HandleFunc("/gshealthz", func(w http.ResponseWriter, r *http.Request) {
+		s.gameServerContainerStarted(initialDelay)
```

Comment:
I think I replied to that in https://github.com/googleforgames/agones/issues/2966#issuecomment-1478300425, but I am now realizing I missed part of your point.

The issue is that right now the SDK starts healthchecking immediately, running this function periodically: https://github.com/googleforgames/agones/blob/main/pkg/sdkserver/sdkserver.go#L820-L826

We could do something simple here and change it so that the `s.healthy()` clause is only even checked if we've seen `/gshealthz`, e.g. `/gshealthz` setting a flag that the container has started. You're right that that might be simpler.

Reply:
It might be possible to use a `sync.Once()` to handle the initial starting operation as well.

But yes, that's exactly my thought 👍🏻

Reply:
> It might be possible to use a sync.Once() to handle the initial starting operation as well.

Yeah, I'm toying with this for sure.

Reply:
PTAL!